packages/echo-core/src/services/analytics/errorToExceptionTelemetry.test.ts

```diff
@@ -0,0 +1,114 @@
+import { BaseError } from '@equinor/echo-base';
+import { errorToExceptionTelemetry } from './errorToExceptionTelemetry';
+
+const defaultArgs = {
+    sessionKey: 'key',
+    moduleName: 'moduleName',
+    instCode: 'JSV',
+    userCompany: 'company'
+};
+
+describe('errorToExceptionTelemetry', () => {
+    it(`ExceptionTelemetry should log properties including staticErrorProperties`, () => {
+        // setup
+        const message = 'error message';
+        const name = 'errorName';
+        const outerError = new BaseError({
+            name,
+            message
+        });
+        outerError.stack = 'ignore';
+        outerError.errorTraceId = 'ignore';
+
+        const staticErrorProperties = { staticProp1: 1, staticProp2: 'prop2' };
+
+        //when
+        const actual = errorToExceptionTelemetry({ error: outerError, ...defaultArgs, staticErrorProperties });
+
+        //then
+        const expected = {
+            exception: outerError,
+            properties: {
+                ...defaultArgs,
+                errorType: name,
+                name,
+                message,
+                stack: 'ignore',
+                errorTraceId: 'ignore',
+                ...staticErrorProperties
+            },
+            severityLevel: 3
+        };
+        expect(actual).toEqual(expected);
+    });
+    it(`ExceptionTelemetry should log all all properties of error and 2 nested inner Errors of different types`, () => {
+        // setup
+        const nestedInnerError2 = new CustomInnerError('inner 2 message');
+        nestedInnerError2.stack = 'ignore';
+
+        const nestedInnerError1 = new CustomBaseError({ message: 'inner 1 message', error: nestedInnerError2 });
+        nestedInnerError1.stack = 'ignore';
+        nestedInnerError1.errorTraceId = 'ignore';
+
+        const outerError = new BaseError({
+            name: 'BaseError',
+            message: 'outer message',
+            innerError: nestedInnerError1
+        });
+        outerError.stack = 'ignore';
+        outerError.errorTraceId = 'ignore';
+        //when
+        const actual = errorToExceptionTelemetry({ error: outerError, ...defaultArgs });
+
+        //then
+        const expected = {
+            exception: outerError,
+            properties: {
+                ...defaultArgs,
+                errorType: 'BaseError',
+                name: 'BaseError',
+                message: 'outer message',
+                stack: 'ignore',
+                errorTraceId: 'ignore',
+
+                innerError: {
+                    customBaseField: 'public1',
+                    customBasePrivateField: 'private1',
+                    message: 'inner 1 message',
+                    name: 'CustomBaseError',
+                    stack: 'ignore',
+                    errorTraceId: 'ignore',
+
+                    innerError: {
```

Comment:
Oh wow, so what does this nested `innerError` mean? This is like an artificial stack trace? 🤔

Reply:
Often we want to decorate 3rd party exception with additional information, like a pdfError with the docNo that failed. But we would still want to be able to see the original error, which will become the innerError.
Something like this:
```
try
{
  renderCanvas(pdf);
} catch(error)
{
  throw new PdfError("Failed to render canvas", docNo, innerError: error);
}
```

Reply:
Ah, my comment isn't precise enough: git doesn't show the full context here of the code.

What I meant under nested innerError is actually a child innerError of another one, in this file.
Full context:
```
                innerError: {
                    customBaseField: 'public1',
                    customBasePrivateField: 'private1',
                    message: 'inner 1 message',
                    name: 'CustomBaseError',
                    stack: 'ignore',
                    errorTraceId: 'ignore',

                    innerError: {
                        anotherField: 'another value',
                        message: 'inner 2 message',
                        name: 'Inner CustomError',
                        stack: 'ignore'
                    }
                    ...
```
So what does this mean here? (the parent innerError + child innerError)

Reply:
It's the same as I described above, just one more level. For example in the search, I catch a Dexie Error, and rethrow it as searchError with dexieError as innerError. And in echopedia we might catch it, and rethrow that again as echopediaError with searchError as innerError (which already has dexieError as innerError)